Scenarios/AKS-Secure-Baseline-PrivateCluster/Terraform/05-Network-LZ/aks-policy.tf

```diff
@@ -0,0 +1,301 @@
+resource "azurerm_resource_group_policy_assignment" "aks_internal_lb" {
+  name                 = "AKS should use internal load balancers"
+  resource_group_id    = azurerm_resource_group.spoke-rg.id
+  policy_definition_id = "/providers/Microsoft.Authorization/policyDefinitions/3fc4dc25-5baf-40d8-9b05-7fe74c1bc64e"
+  location             = "Global"
+
+  parameters = <<PARAMS
+    {
+      "excludedNamespaces": {
+        "value": []
+      },
+      "effect": {
+        "value": "Deny"
+      }
+    }
+PARAMS
+}
+
+#  Built-in 'Kubernetes cluster containers should run with a read only root file system' Azure Policy for Kubernetes policy definition
+resource "azurerm_resource_group_policy_assignment" "aks_ro_fs" {
+  name                 = "AKS containers should run with a read only root file system"
+  resource_group_id    = azurerm_resource_group.spoke-rg.id
+  policy_definition_id = "/providers/Microsoft.Authorization/policyDefinitions/df49d893-a74c-421d-bc95-c663042e5b80"
+  location             = "Global"
+
+  parameters = <<PARAMS
+    {
+      "excludedNamespaces": {
+        "value": [
+          "kube-system",
+          "gatekeeper-system",
+          "azure-arc",
+          "flux-system",
+          "windows-gmsa-webhook-system",
+          "simpleapp"
+        ]
+      },
+      "excludedContainers": {
+        "value": [
+          "kured"
+        ]
+      },
+      "effect": {
+        "value": "Deny"
+      }
+    }
+PARAMS
+}
+
+#  Built-in 'AKS container CPU and memory resource limits should not exceed the specified limits
+resource "azurerm_resource_group_policy_assignment" "aks_cpu_mem_limit" {
+  name                 = "AKS CPU and memory resource limits should not exceed"
+  resource_group_id    = azurerm_resource_group.spoke-rg.id
+  policy_definition_id = "/providers/Microsoft.Authorization/policyDefinitions/e345eecc-fa47-480f-9e88-67dcc122b164"
+  location             = "Global"
+
+  parameters = <<PARAMS
+    {
+      "cpuLimit": {
+        "value": "500m"
+      },
+      "memoryLimit": {
+        "value": "1024Mi"
+      },
+      "excludedNamespaces": {
+        "value": [
+          "kube-system",
+          "gatekeeper-system",
+          "azure-arc",
+          "flux-system",
+          "windows-gmsa-webhook-system"
+        ]
+      },
+      "effect": {
+        "value": "Deny"
+      }
+    }
+PARAMS
+}
+
+#  Built-in 'AKS containers should only use allowed images'
+resource "azurerm_resource_group_policy_assignment" "aks_containers_allowed" {
+  name                 = "AKS containers should only use allowed images"
+  resource_group_id    = azurerm_resource_group.spoke-rg.id
+  policy_definition_id = "/providers/Microsoft.Authorization/policyDefinitions/febd0533-8e55-448f-b837-bd0e06f16469"
+  location             = "Global"
+
+   parameters = <<PARAMS
+    {
+      "allowedContainerImagesRegex": {
+        "value": "mcr.microsoft.com"
```

Comment:
Yeah, because acr is built after the policies and uses random number generator for name. I could move policy.tf to acr build under 06-AKS-supporting step or hard code the acr to be predictable. Which one do you think? I agree it was weighing on my mind.